### PR TITLE
nixos/ibus: don't use xdg autostart for Gnome

### PR DESCRIPTION
**Pull request by jian-lin**
###### Description of changes

Gnome uses systemd to start ibus.

This depends on https://github.com/NixOS/nixpkgs/pull/176876 to patch gnome's ibus systemd service.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jian-lin** [comment]
@ttuegel  @jtojnar

**Comment by jtojnar** [comment]
I pushed the fix for the services.

I verified in VM that the systemd service is properly started and that it points to the wrapped daemon. I also checked that I can add Mozc in the Control Center.

File: pkgs/tools/inputmethods/ibus/default.nix [hunk 1]
```diff
@@ -34,6 +34,7 @@
 , buildPackages
 , runtimeShell
 , nixosTests
+, ibus
```
Comment (jian-lin) [inline pkgs/tools/inputmethods/ibus/default.nix:37]: Is it needed? @jtojnar

Comment (jtojnar) [reply pkgs/tools/inputmethods/ibus/default.nix]: Thanks, that was a remnant of previous attempt.

**Comment by jian-lin** [comment]
I tested what @jtojnar says on my laptop, and it works well.

 Thanks @jtojnar

**Pull request merged**